mimic/core.py

```diff
@@ -128,9 +153,17 @@ def entries_for_tenant(self, tenant_id, prefix_map, base_uri):
         :return: The full URI locating the service for that region
         """
         for service_id, api in self._uuid_to_api.items():
-            for entry in api.catalog_entries(tenant_id):
-                for endpoint in entry.endpoints:
-                    prefix_map[endpoint] = self.uri_for_service(
-                        endpoint.region, service_id, base_uri
-                    )
-                yield entry
+            if IExternalAPIMock.providedBy(api):
+                for entry in api.catalog_entries(tenant_id):
+                    for endpoint in entry.endpoints:
+                        prefix_map[endpoint] = api.uri_for_service(
+                            endpoint.region, service_id
+                        )
+                    yield entry
+            else:  # IAPIMock.providedBy(api)
```

Comment:
> Not that I'm expecting a lot of external mocks, but it can reduce branching if the storage is split

Exactly, I'm just looking to simplify the code, not address issues of scale.